### ISSUE TITLE
Error while running query with ANY clause

### ISSUE THREAD

**Issue by samay-sharma** [citusdata/citus#443]
Running a query like below leads to an error in Citus

``` sql
SELECT * FROM test_table WHERE a = ANY(VALUES (1),(2),(3));
ERROR:  cannot perform distributed planning on this query
DETAIL:  Complex table expressions are currently unsupported
```

The same query works on a regular PostgreSQL table.

**Comment by samay-sharma** [comment]
The query works if it is written like this:

```sql
SELECT * from test_table where a = ANY(ARRAY[1,2,3]);
```

**Comment by ozgune** [comment]
I'm adding the following conversation from #991 below.

---
Is it possible to integrate support of ``any`` values to citus?

With Citus 6 I get the following error:

>> Error: cannot perform distributed planning on this query

I could use ``IN`` or ``ANY(ARRAY... )`` expression, but that way is much slower. Seems like there are similar experiences/requests in issue: #443 and #779

It's a postgres related problem, that the queryplanner is kinda slow with ``IN()`` and the best way I found to get around this is ``ANY VALUES``.

``ANY VALUE`` query runs 3 seconds, ``IN / ANY(ARRAY )`` runs up to 2 minutes. We have 8000 - 12000 values in the filter clauses.

**Comment by velioglu** [comment]
Closed via #1475
